lxd/storage/backend_lxd.go

```diff
@@ -3246,9 +3246,8 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 				return err
 			}
 
-			// Reset img volume variables as we just deleted the old one.
+			// Reset img volume variable as we just deleted the old one.
 			imgDBVol = nil
-			imgVol = b.GetVolume(drivers.VolumeTypeImage, contentType, fingerprint, nil)
```

Comment:
Can you explain more about this change, it doesn't look right to me as we just deleted the volume on line 3244

Reply:
Can you give an example? 

I'm worried that this change affects all storage drivers, does the bug your fixing also affect other storage drivers or only zfs with block mode enabled?

Reply:
If we kept this line, calling `b.driver.CreateVolume` will be passed a volume without a config. Both `ceph` and `zfs` compare the values of the physical volume size and the size defined in the config. Now, this will always fail if we don't pass a volume config.

Reply:
> I'm worried that this change affects all storage drivers, does the bug your fixing also affect other storage drivers or only zfs with block mode enabled?

It should only affect ceph and zfs, as those are the only storage drivers which keep cached images around.